src/bidders/index.js

```diff
@@ -0,0 +1,154 @@
+import { context } from '@wikia/ad-engine';
+import { A9 } from './a9/index';
+import { Prebid } from './prebid/index';
+
+
+require('./../../lib/prebid.min');
+
+const bidIndex = {
+	a9: {
+		pos: 2,
+		char: '9'
+	},
+	prebid: {
+		pos: 4,
+		char: 'P'
+	}
+};
+
+let bidderMarker = ['x', 'x', 'x', 'x', 'x'];
+const biddersRegistry = {};
+const realSlotPrices = {};
+
+function applyTargetingParams(slotId, targeting) {
+	Object
+		.keys(targeting)
+		.forEach(
+			key => context.set(`slots.${slotId}.targeting.${key}`, targeting[key])
```

Comment:
What happens when we refresh the slot/load multiple slots and bid was already rendered? What happens on consecutive pvs on mobile-wiki? Should we clear bid states after render/pv change?

Reply:
On every non-lazy call adUnits are gonna be removed as well as bids responses and if mobile-wiki pageChange event will be rose then also bidders state will be reseted. If you ask about FMR X then at this moment with different slot names (increasing number) second and later call will probably ignore bidders and I think that this should be fixed in FMR logic. Follow up ticket maybe?